src/metabase/driver/sql/query_processor/deprecated.clj

```diff
@@ -0,0 +1,128 @@
+  `::metabase.query-processor.util.add-alias-info/desired-alias` information in the
+  `:field`/`:expression`/`:aggregation` options map. See [[metabase.query-processor.util.add-alias]] for more
+  information.
+
+  Drivers that need to customize the aliases used can override
+  the [[metabase.query-processor.util.add-alias-info/escape-alias]] multimethod, or change the values of
+  `::metabase.query-processor.util.add-alias-info/desired-alias` or
+  `::metabase.query-processor.util.add-alias-info/source-alias` in the appropriate [[->honeysql]] methods."
+  {:arglists '([driver field]), :deprecated "0.41.0"}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmethod field->alias :sql
+  [driver [_ _ {::add/keys [desired-alias]} :as field]]
+  (log-deprecation-warning driver 'metabase.driver.sql.query-processor/field->alias "v0.41.0")
+  (or desired-alias
+      (driver/escape-alias driver (:name field))))
+
+
+;;; DEPRECATED IN 42
+
+(def ^:dynamic ^{:deprecated "0.42.0"} *field-options*
+  "DEPRECATED (unused) in 0.42.0. Binding this var has no effect.
+
+  If you need to track `:field` options for one reason or another, you can create and bind your own dynamic variable
+  for doing so. See [[metabase.driver.sqlserver]] for an example.
+
+  This var will be removed in a future release."
+  nil)
+
+(def ^:dynamic ^{:deprecated "0.42.0"} *table-alias*
+  "The alias, if any, that should be used to qualify Fields when building the HoneySQL form, instead of defaulting to
+  schema + Table name. Used to implement things like joined `:field`s.
+
+  DEPRECATED (unused) in 0.42.0+. Instead of using this, use or override `::add/source-table` in the
+  `:field`/`:expression`/`:aggregation` options."
+  nil)
+
+(def ^:dynamic ^{:deprecated "0.42.0"} *source-query*
+  "The source-query in effect.  Used when the query processor might need to distinguish between the type of the source
+  query (ex: to provide different behavior depending on whether the source query is from a table versus a subquery).
+
+  DEPRECATED -- use [[*query*]] instead, which does the same thing but it always bound even if we are not in a source
+  query."
+  nil)
+
+(defmulti field->identifier
+  "DEPRECATED: Unused in 0.42.0+; this functionality is now handled by [[->honeysql]]. Implementing this method has no
+  effect. This method will be removed in a future release."
+  {:arglists '([driver field]), :deprecated "0.42.0"}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmulti escape-alias
+  "DEPRECATED -- this has been moved to [[driver/escape-alias]]."
+  {:added "0.41.0", :deprecated "0.42.0", :arglists '([driver column-alias-name])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
+(defmethod escape-alias :sql
+  [driver column-alias-name]
+  ((get-method driver/escape-alias ::driver/driver) driver column-alias-name))
+
+(defmethod driver/escape-alias :sql
```

Comment:
It seems weird to declare this here.  Shouldn't it be in the driver namespace directly?

Reply:
The only reason this method implementation is needed at all is because it's calling the deprecated `escape-alias` method above (if a method impl exists for it). Once we remove `escape-alias` this method impl can be removed along with it.

Since it's not really a default implementation for `:sql` so much as it is a little bit of magic to us the deprecated method if it's still around, it seemed like it was appropriate to put it in with the rest of the deprecated stuff. I'll add a note to clarify that tho.